lib/Conversion/TorchToTcp/DataMovement.cpp

```diff
@@ -28,21 +29,22 @@ using namespace mlir::torch;
 using namespace mlir::torch::Torch;
 
 namespace {
-// Following function is copied from
-// https://sourcegraph.com/github.com/llvm/torch-mlir@4446fa00d8258311867496fc79d0b1dddd22a972/-/blob/lib/Conversion/TorchToLinalg/DataMovement.cpp?L42
-// TODO: Expose this function in a header to reuse.
-LogicalResult prepareArgumentsForSlicingOp(AtenSliceTensorOp op,
-                                           AtenSliceTensorOpAdaptor adaptor,
+// Adopted from
+// https://sourcegraph.com/github.com/llvm/torch-mlir@6b3a7d07c2c76f5e8437ff4e88110899621557b9/-/blob/lib/Conversion/TorchToLinalg/DataMovement.cpp?L42
+template <typename OpTy, typename OpAdaptor>
+LogicalResult prepareArgumentsForSlicingOp(OpTy op, OpAdaptor adaptor,
```

Comment:
There were new updates to this function upstream - I've updated it here, and added the templatization back to make it easier to sync with upstream going forward.